src/v/kafka/server/tests/create_topics_test.cc

```diff
@@ -565,3 +566,37 @@ FIXTURE_TEST(unlicensed_reject_defaults, create_topic_fixture) {
         update_cluster_config(config, "false");
     }
 }
+
+FIXTURE_TEST(create_dry_run_rejects_existing, create_topic_fixture) {
+    auto client = make_kafka_client().get();
+    client.connect().get();
+
+    auto topic = make_topic(ssx::sformat("topic_foo"));
+
+    // create the topic
+    auto resp = client
+                  .dispatch(
+                    make_req({topic}, /*validate_only = */ false),
+                    kafka::api_version(5))
+                  .get();
+    BOOST_REQUIRE_EQUAL(
+      resp.data.topics[0].error_code, kafka::error_code::none);
+
+    // rejects attempt to create it again
+    resp = client
+             .dispatch(
+               make_req({topic}, /*validate_only = */ false),
+               kafka::api_version(5))
+             .get();
+    BOOST_REQUIRE_EQUAL(
+      resp.data.topics[0].error_code, kafka::error_code::topic_already_exists);
+
+    // also rejects dry run create
+    resp = client
+             .dispatch(
+               make_req({topic}, /*validate_only = */ true),
+               kafka::api_version(5))
+             .get();
+    BOOST_REQUIRE_EQUAL(
+      resp.data.topics[0].error_code, kafka::error_code::topic_already_exists);
```

Comment:
for a dry-run feature, i would think the test should also try to create the topic with the dry-run flag set, and observe that the topic wasn't created?

Reply:
maybe there are other cases too?

Reply:
There are other dry run tests. Not totally sure how coverage is...i can check. This is for the specific case where the topic exists